log/klogv2/klogv2.go

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2021 The virtual-kubelet authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package klogv2 implements a virtual-kubelet/log.Logger using klogv2 as a backend
+//
+// You can use this by creating a klogv2 logger and calling `FromKlogv2(fields)`.
+// If you want this to be the default logger for virtual-kubelet, set `log.L` to the value returned by `FromKlogv2`
+//
+// We recommend reading the klog conventions to build a better understanding of levels and when they should apply
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md
+package klogv2
+
+import (
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"k8s.io/klog/v2"
+)
+
```

Comment:
What do you think about doing a:

```go
var _ log.Logger = (*adapter)(nil)
```

Reply:
(rather than the test in the other file)

Reply:
Just following what was there but yeah I'm all in for compile time validation. On it.